src/text/raw_text_reader.rs

```diff
@@ -91,10 +91,16 @@ impl<T: TextIonDataSource> RawTextReader<T> {
             }
 
             // Otherwise, see if the next token in the stream is an Ion Version Marker.
-            if let Ok(Some(_)) = self.parse_next(ion_1_0_version_marker) {
-                // We found an IVM; we currently only support Ion 1.0.
-                self.current_ivm = Some((1, 0));
-                return Ok(());
+            match self.parse_next(ion_1_0_version_marker) {
+                Ok(Some(_)) => {
+                    // We found an IVM; we currently only support Ion 1.0.
+                    self.current_ivm = Some((1, 0));
+                    return Ok(());
+                }
+                Err(e @ IonError::IoError { .. }) => return Err(e),
```

Comment:
🗺️ Previously we would ignore any kind of `Err(_)` here. In most cases this was fine because we expect parsers like `ion_1_0_version_marker` to return an `Err` if they don't match the input. That's both common and expected.

If we encountered an `IoError`, however, it means there was a problem caused by something other than parsing. It could be a failure to read from the disk or a socket, or it could mean that the input contained invalid utf-8 data. We need to surface this to the application.